README.md

```diff
@@ -1,2 +1,140 @@
+![Spectre](https://user-images.githubusercontent.com/1897842/31115297-0fe2c3aa-a822-11e7-90e6-92ceccf76137.jpg)
+
 # analysis-api
-Worker for manipulation of analyses.
+
+API for accessing details of analyses, their configurations and scheduling. Its
+main feature is high extensibility of the system through automated discovery
+and integration of new analysis modules.
+
+# Specification of a Worker
+
+In order to make worker discoverable and useful in the system, following
+specification must be met.
+
+## Input Specification Endpoint
+
+Endpoint `/schema/inputs/<task-name>` should provide simplistic specification
```

Comment:
What about automatically generated list of tasks based on available workers?

Reply:
This is handled on `master`-node side.